packages/fxa-payments-server/src/lib/validator.ts

```diff
@@ -0,0 +1,155 @@
+import { useReducer, useMemo } from 'react';
```

Comment:
This module implements a reducer-based data store for overall form validation state - the `useReducer` hook causes a re-render of the form with every dispatched change to the state. And the `Validator` class works as a utility for working with the store in form field components.

Reply:
I guess this is because we're not using component-local state?